<launch>

    <!-- <param name="use_sim_time" value="false"/> -->


    <arg name="fixed_frame" default="sg_map"/>

    <!-- <param name="robot_description" textfile="$(find hybrid_a_star)/model/car_model/default.urdf"/> -->
    <!-- <node pkg="joint_state_publisher" type="joint_state_publisher" name="joint_state_publisher"/> -->
    <!-- <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher"/> -->

    <node pkg="rviz" type="rviz" name="rviz_Astar" args="-d $(find hybrid_a_star)/rviz/display_hybrid_astar.rviz"/>

    <!--static_transform_publisher x y z yaw pitch roll frame_id child_frame_id period_in_ms-->
    <node pkg="tf" type="static_transform_publisher" name="link_broadcaster" args="-25 -25 0 0 0 0 localMap sg_map 1000"/>

    <node pkg="hybrid_a_star" type="run_hybrid_astar" name="run_hybrid_astar" output="screen">
    
        <param name="use_sim_time" value="true"/>

        <param name="planner/steering_angle" value="35.0"/>
        <param name="planner/steering_angle_discrete_num" value="3"/>
        <param name="planner/wheel_base" value="2.88"/>
        <param name="planner/segment_length" value="3.2"/>
        <param name="planner/segment_length_discrete_num" value="4"/>
        <param name="planner/steering_penalty" value="1.5"/>
        <param name="planner/reversing_penalty" value="100.0"/>
        <param name="planner/steering_change_penalty" value="2"/>
        <param name="planner/shot_distance" value="5.0"/>
    </node>

</launch>
